app/api/api_v1/endpoints/submission_router.py

```diff
@@ -59,38 +69,33 @@ async def get_own_submissions(
     assignment_id: int
 ):
     onyen = request.user.onyen
+    submission_service = SubmissionService(db)
     student = await StudentService(db).get_user_by_onyen(onyen)
     assignment = await AssignmentService(db).get_assignment_by_id(assignment_id)
-    submissions = await SubmissionService(db).get_submissions(student, assignment)
+    submissions = await submission_service.get_submissions(student, assignment)
 
-    return submissions
+    return [await submission_service.get_submission_schema(s) for s in submissions]
 
-@router.get("/latest_submission/", response_model=SubmissionSchema)
-async def get_latest_submission(
+@router.get("/active_submission/", response_model=SubmissionSchema)
```

Comment:
These endpoint URIs seem like they need to be cleaned up. I feel like this one should be `GET /submissions/active` and the one below on line 126 should be `GET /submissions/active/download` (well, I think they should be completely different and use path params instead of query params, but this would clean them up with minimal changes).

Reply:
fixing this